pkg/datastore/datastore.go

```diff
@@ -543,6 +543,11 @@ func (wo WatchOptions) WithCheckpointInterval(interval time.Duration) WatchOptio
 
 // ReadOnlyDatastore is an interface for reading relationships from the datastore.
 type ReadOnlyDatastore interface {
+	// UniqueID returns a unique identifier for the datastore. This identifier
+	// must be stable across restarts of the datastore if the datastore is
+	// persistent.
+	UniqueID(context.Context) (string, error)
```

Comment:
Did you consider alternatives using the `UniqueID` as a stable identifier for zedtokens?

I _assume_ (not present in the PR body) that the goal is making sure zedtokens from one datastore _type_ are not used in another datastore _type_. But what happens if you evolve the zedtoken implementation from one version to another for the same datastore, and want to force them to be dropped? the `uniqueID` wouldn't help, would it? 
Shouldn't we store a zedtoken versioning parameter like `datastoreType+zedtokenVersion`?

Reply:
Yes, but then it doesn't meet the goal which is to prevent zedtokens not just across *types* but *instances* as well

Reply:
>But what happens if you evolve the zedtoken implementation from one version to another for the same datastore, and want to force them to be dropped? the uniqueID wouldn't help, would it?
Shouldn't we store a zedtoken versioning parameter like datastoreType+zedtokenVersion?

Can you address this ☝️?